src/strategy/GGVStrategy.sol

```diff
@@ -9,184 +12,245 @@ import {WithdrawalQueue} from "src/WithdrawalQueue.sol";
+            .call(
+                address(TELLER),
+                abi.encodeWithSelector(
+                    TELLER.deposit.selector, address(WSTETH), _wstethToMint, params.minimumMint, _referral
+                )
+            );
+        uint256 ggvShares = abi.decode(data, (uint256));
+
+        emit StrategySupplied(msg.sender, _referral, msg.value, stv, _wstethToMint, _params);
+        emit GGVDeposited(msg.sender, _wstethToMint, ggvShares, _referral, _params);
     }
 
-    /// @notice Requests a withdrawal of ggv shares from the strategy
-    /// @param _stethSharesToBurn The amount of steth shares to burn
-    /// @param _params The parameters for the withdrawal
-    /// @return requestId The request id
-    function requestExitByStethShares(uint256 _stethSharesToBurn, bytes calldata _params)
-        public
-        returns (bytes32 requestId)
-    {
-        GGVParams memory params = abi.decode(_params, (GGVParams));
+    // =================================================================================
+    // REQUEST EXIT FROM STRATEGY
+    // =================================================================================
+
+    /**
+     * @notice Previews the amount of wstETH that can be withdrawn by a given amount of GGV shares
+     * @param _ggvShares The amount of GGV shares to preview the amount of wstETH for
+     * @param _params The parameters for the withdrawal
+     * @return wsteth The amount of wstETH that can be withdrawn
+     */
+    function previewWstethByGGV(uint256 _ggvShares, bytes calldata _params) public view returns (uint256 wsteth) {
+        // uint256 -> uint128 check
+        GGVParamsRequestExit memory params = abi.decode(_params, (GGVParamsRequestExit));
+        wsteth = BORING_QUEUE.previewAssetsOut(address(WSTETH), uint128(_ggvShares), params.discount);
+    }
 
-        address callForwarder = _getOrCreateCallForwarder(msg.sender);
+    /**
+     * @inheritdoc IStrategy
+     */
+    function requestExitByWsteth(uint256 _wsteth, bytes calldata _params) external returns (bytes32 requestId) {
+        GGVParamsRequestExit memory params = abi.decode(_params, (GGVParamsRequestExit));
+
+        IStrategyCallForwarder callForwarder = _getOrCreateCallForwarder(msg.sender);
         IERC20 boringVault = IERC20(TELLER.vault());
 
         // Calculate how much wsteth we'll get from total GGV shares
-        uint256 totalGGV = boringVault.balanceOf(callForwarder);
-        uint256 totalStethSharesFromGgv =
-            BORING_QUEUE.previewAssetsOut(address(WSTETH), uint128(totalGGV), params.discount);
-        if (totalStethSharesFromGgv == 0) revert InvalidStethAmount();
-        if (_stethSharesToBurn > totalStethSharesFromGgv) revert InvalidStethAmount();
+        uint256 totalGGV = boringVault.balanceOf(address(callForwarder));
+        uint256 totalWstethFromGGV = previewWstethByGGV(totalGGV, _params);
+        if (totalWstethFromGGV == 0) revert InvalidWstethAmount();
```

Comment:
Maybe smth like this? Semantically it's a bit different situation

InvalidWstethAmount -> NothingToExit or NothingToWithdraw